modules/configuration/src/main/java/org/apache/ignite/internal/configuration/asm/ConfigurationAsmGenerator.java

```diff
@@ -1324,7 +1326,16 @@ private void addNodeConstructMethod(
                     .ret();
         }
     }
-    
+
+    private void makeSureChangePolymorphicTypeIdMethodIsDefined(@Nullable MethodDefinition changePolymorphicTypeIdMtd,
+                                                                Field schemaField) {
+        if (changePolymorphicTypeIdMtd == null) {
```

Comment:
It is correct to check for the schema itself, not the field.

Reply:
`schemaField` is only passed to make assertion message more informative, we are not asserting anything on the field itself.